### DISCUSSION TITLE
Error in TS with Ref in assing prop with Loaded prop

### DISCUSSION THREAD

**Discussion by Nisgrak** [mikro-orm/mikro-orm#4493] [Q&A] [answered]
Hi!

I'm starting to use Ref and have one problem with TS types.

I have an app with Issues, Cells and CellsLocation and want to create an issue and assign a cell with the cellLocation loaded but TS gives an error.

What would be the actual way to assign?

I create a repro, Issue -> User, FirstSublevel -> Cell, SecondSublevel -> CellLocation to continue with my actual problem:
https://stackblitz.com/edit/mikro-orm-esm-repro-c9uvgo?file=basic.test.ts

**Comment by B4nan** [comment, marked as answer]
This is already resolved in v6.

You can also use `em.assign` which will handling wrapping references automatically, e.g. `em.assign(newUser, { firstSublevel: test })`.

**Reply by Nisgrak** [reply]
Thanks!! For the v6 you have more updates to let know? The last "date update" in the discussion was in march
